build.sh

```diff
@@ -52,17 +52,22 @@ function build_fluentbit() {
   cd submodules/fluent-bit
   mkdir -p build
   cd build
-  cmake .. -DCMAKE_INSTALL_PREFIX=$subagentdir/fluent-bit -DFLB_HTTP_SERVER=ON -DFLB_DEBUG=OFF -DCMAKE_BUILD_TYPE=RelWithDebInfo -DWITHOUT_HEADERS=ON
+  cmake .. -DCMAKE_INSTALL_PREFIX=/usr/lib/google-cloud-ops-agent \
+    -DCMAKE_INSTALL_BINDIR=. -DFLB_HTTP_SERVER=ON -DFLB_DEBUG=OFF \
+    -DCMAKE_BUILD_TYPE=RelWithDebInfo -DWITHOUT_HEADERS=ON -DFLB_SHARED_LIB=OFF
   make -j8
   make DESTDIR="$DESTDIR" install
-  # We don't want fluent-bit's service
-  rm "$DESTDIR/lib/systemd/system/fluent-bit.service"
-  rm -r "${DESTDIR}${subagentdir}/fluent-bit/etc"
+  # We don't want fluent-bit's service or configuration, but there are no cmake
+  # flags to disable them. Prune after build.
+  rm -r "${DESTDIR}/usr/lib/google-cloud-ops-agent/etc"
```

Comment:
Does it make sense to force `fluent-bit` into the path and only remove those specific paths? e.g.

```
rm -r "${DESTDIR}/usr/lib/google-cloud-ops-agent/fluent-bit/etc"
rm -r "${DESTDIR}/lib/fluent-bit"
```

The potential benefit is to avoid accidentally removing other stuff that happens to be in the same paths.

e.g. Can we do `-DCMAKE_INSTALL_PREFIX=/usr/lib/google-cloud-ops-agent/fluent-bit` above?

Reply:
if we set `-DCMAKE_INSTALL_PREFIX=/usr/lib/google-cloud-ops-agent/fluent-bit` the final paths will be e.g. /usr/lib/google-cloud-ops-agent/fluent-bit/fluent-bit, /usr/lib/google-cloud-ops-agent/fluent-bit/etc/fluent-bit, and /lib (the systemd install paths are not affected by the prefix variable). so in this case we'd need to do more rm's and also a 'mv'. i set the prefix to /usr/lib/google-cloud-ops-agent because that's where we want the one thing we are keeping, so i optimized for that file. let me know what you think

Reply:
Oh I see. Can we keep the path unchanged, but delete by `/usr/lib/google-cloud-ops-agent/fluent-bit`, `/usr/lib/google-cloud-ops-agent/etc/fluent-bit`? Is there anything else in `/lib` that is not Fluent Bit specific, and might get deleted by accident?

Reply:
As build.sh is written, the fluent-bit build happens first, so only fluent-bit files are in this directory. I don't follow your alternative suggestion here, could you highlight and make a suggested change in the PR?

Reply:
I meant something like changing

```
rm -r "${DESTDIR}/usr/lib/google-cloud-ops-agent/etc"
rm -r "${DESTDIR}/lib/"
```

to

```
rm -r "${DESTDIR}/usr/lib/google-cloud-ops-agent/fluent-bit"
rm -r "${DESTDIR}/usr/lib/google-cloud-ops-agent/etc/fluent-bit"
rm -r "${DESTDIR}/lib/systemd/system/fluent-bit.service" // Not sure exactly what the path is here
```

I couldn't find a way to add `suggested changes`. I guess Github does not support it yet.

Reply:
if we change to  `-DCMAKE_INSTALL_PREFIX=/usr/lib/google-cloud-ops-agent/fluent-bit`
then we would run the commands
```
mv "${DESTDIR}/usr/lib/google-cloud-ops-agent/fluent-bit/fluent-bit" "${DESTDIR}/usr/lib/google-cloud-ops-agent/fluent-bit"
rm -r "${DESTDIR}/usr/lib/google-cloud-ops-agent/fluent-bit"
rm -r "${DESTDIR}/lib"
```

this seemed messier but if you want i will update to this

Reply:
Yeah, that's a bit messier. Sounds like there is no way to avoid `rm -r "${DESTDIR}/lib"` any way. Might as well leave as is then.